### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by abhro**
- Add DLMF reference identifiers
- Add sections to API list
- Clean up docstrings

### CODE REVIEW

File: docs/src/functions_list.md [hunk 1]
```diff
@@ -4,7 +4,156 @@
...
+
+```@docs
+eta
+zeta
+```
+
+## Misc
+
+```@docs
+_cotpi
```
Comment (giordano) [inline docs/src/functions_list.md:158]: This sounds like an internal function which shouldn't be exposed in public API

File: src/sincosint.jl [hunk 2]
```diff
@@ -233,12 +233,13 @@ Compute the sine integral function of ``x``, defined by
 
 ```math
 \operatorname{Si}(x)
-:= \int_0^x \frac{\sin t}{t} \, \mathrm{d}t
+:= \int_0^x \frac{\sin t}{t} \, dt
```
Comment (giordano) [inline src/sincosint.jl:236]: Why? `\mathrm{d}t` looks better to me.

Comment (abhro) [reply src/sincosint.jl:236]: This is just for consistency with some other integral formulas which don't use `\mathrm`.

File: docs/src/functions_list.md (hunk 1 above)
Comment (abhro) [reply docs/src/functions_list.md:158]: Agreed. The `@autodocs` block exposed it, so I was keeping it consistent. Will remove it. Thank you!

File: src/sincosint.jl (hunk 2 above)
Comment (giordano) [reply src/sincosint.jl:236]: Then why not changing the other integral formulas which don't use `\mathrm`?

Comment (abhro) [reply src/sincosint.jl:236]: Fair point. It was really a swing either way over which form to use, and it makes just as much sense to do the `\mathrm`. Will be pushing that soon. Thank you!

**Comment by ViralBShah** [comment]
This broadly looks great. Merging it.

**Pull request merged**